pwiz_tools/Skyline/Model/IonMobility/IonMobilityLibraryManager.cs

```diff
@@ -50,7 +50,8 @@ public override void ClearCache()
 
         protected override bool StateChanged(SrmDocument document, SrmDocument previous)
         {
-            return !ReferenceEquals(GetIonMobilityLibrary(document), GetIonMobilityLibrary(previous));
+            return !ReferenceEquals(GetIonMobilityLibrary(document), GetIonMobilityLibrary(previous))
+                   || !ReferenceEquals(document.Id, previous?.Id);

```

Comment:
Should this go up a level into the calling logic that checks for a state change? Do we ever want to consider changing the document ID as potentially not a state change?